nrf-hal-common/src/twis.rs

```diff
@@ -0,0 +1,322 @@
+//! HAL interface to the TWIS peripheral.
+//!
+
+use core::ops::Deref;
+use core::sync::atomic::{compiler_fence, Ordering::SeqCst};
+
+use crate::pac::{twis0, P0, TWIS0};
+
+#[cfg(any(feature = "52832", feature = "52833", feature = "52840"))]
+use crate::pac::TWIS1;
+
+use crate::pac::{
+    generic::Reg,
+    twis0::{_EVENTS_READ, _EVENTS_STOPPED, _EVENTS_WRITE, _TASKS_STOP},
+};
+
+use crate::{
+    gpio::{Floating, Input, Pin},
+    slice_in_ram_or,
+    target_constants::EASY_DMA_SIZE,
+};
+
+/// Interface to a TWIS instance.
+pub struct Twis<T>(T);
+
+impl<T> Twis<T>
+where
+    T: Instance,
+{
+    pub fn new(twis: T, pins: Pins, address0: u8) -> Self {
+        // The TWIS peripheral requires the pins to be in a mode that is not
+        // exposed through the GPIO API, and might it might not make sense to
+        // expose it there.
+        //
+        // Until we've figured out what to do about this, let's just configure
+        // the pins through the raw peripheral API. All of the following is
+        // safe, as we own the pins now and have exclusive access to their
+        // registers.
+        for &pin in &[pins.scl.pin(), pins.sda.pin()] {
+            unsafe { &*P0::ptr() }.pin_cnf[pin as usize].write(|w| {
+                w.dir()
+                    .input()
+                    .input()
+                    .connect()
+                    .pull()
+                    .pullup()
+                    .drive()
+                    .s0d1()
+                    .sense()
+                    .disabled()
+            });
+        }
+
+        twis.psel.scl.write(|w| {
+            let w = unsafe { w.pin().bits(pins.scl.pin()) };
+            #[cfg(feature = "52840")]
```

Comment:
Is the 52833 intentionally missing here?

Reply:
Maybe we should make a combined feature flag (e.g. `extended-port`) which is required by all the chips with an extended port that is used for all those places, so we cannot forget any newly added cores (it would also make adding new cores easier).

Reply:
Yeah, will do that at some point

Reply:
Ahh no that was not intentional, good catch! 👍